laa_court_data_api_app/routers/defendants.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import Optional
+from uuid import UUID
+
+from fastapi import APIRouter
+from fastapi.responses import Response
+
+from laa_court_data_api_app.internal.court_data_adaptor_client import CourtDataAdaptorClient
+from laa_court_data_api_app.models.defendants.defendants_response import DefendantsResponse
+from laa_court_data_api_app.models.prosecution_cases.prosecution_cases_results import ProsecutionCasesResults
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.get('/v2/defendants', response_model=DefendantsResponse, status_code=200)
+async def get_defendants(urn: Optional[str] = None,
+                         name: Optional[str] = None,
+                         dob: Optional[str] = None,
+                         uuid: Optional[UUID] = None):
+    client = CourtDataAdaptorClient()
+    logger.info("Calling GET Endpoint")
+
+    if name and dob:
+        logging.info("Defendants_Get_Name_And_Dob_Filtered")
+        cda_response = await client.get("/api/internal/v2/prosecution_cases",
+                                        params={"filter[name]": name, "filter[date_of_birth]": dob})
+    elif urn and uuid:
+        logging.info(f"Defendants_Get_Urn_And_Uuid_{urn}_{uuid}")
+        cda_response = await client.get(f"/api/internal/v2/prosecution_cases/{urn}/defendants/{uuid}")
+    elif urn:
+        logging.info(f"Defendants_Get_Urn_{urn}")
+        cda_response = await client.get("/api/internal/v2/prosecution_cases",
+                                        params={"filter[prosecution_case_reference]": urn})
+    else:
+        logger.error("Invalid_Defendant_Search")
+        return Response(status_code=404)
```

Comment:
Status code would be 400 here. 404 indicates not found where 400 is bad request

Reply:
Ah thanks!